### PR TITLE
rpc: Change getmininginfo to getstakinginfo

### LINKED ISSUE #2392: `getstakinginfo` should be an alias for `getmininginfo`
# Feature Request

**Describe the Feature Request**
In this [commit](https://github.com/gridcoin-community/Gridcoin-Research/commit/40efc64ffec5046aaf7c584bd0fbe61c9ee09c86) from 2017, `getstakinginfo` was changed to be an alias for `getmininginfo`.  As of 5.3.2.0, this does not appear to work anymore:

image

**Describe Preferred Solution**
`getstakinginfo` should return the same RPC output as `getmininginfo`.

### PR DESCRIPTION
**Pull request by jamescowens**
Retain getmininginfo and the mining help category as an alias for backwards compatibility. Closes #2392.

### CODE REVIEW

File: src/rpc/server.cpp [hunk 1]
```diff
@@ -348,20 +348,21 @@ static const CRPCCommand vRPCCommands[] =
     { "walletpassphrasechange",  &walletpassphrasechange,  cat_wallet        },
 
   // Mining commands
```
Comment (barton2526) [inline src/rpc/server.cpp]: should this comment be adjusted?

File: src/rpc/mining.cpp [hunk 2]
```diff
@@ -20,11 +20,11 @@
...
 
-UniValue getmininginfo(const UniValue& params, bool fHelp)
+UniValue getstakinginfo(const UniValue& params, bool fHelp)
 {
     if (fHelp || params.size() != 0)
         throw runtime_error(
-            "getmininginfo\n"
+            "getstakinginfo\n"
             "\n"
             "Returns an object containing mining-related information\n");
```
Comment (barton2526) [inline src/rpc/mining.cpp]: suggestion "Returns an object containing staking-related information\n"

File: src/rpc/server.cpp (hunk 1 above)
Comment (jamescowens) [reply src/rpc/server.cpp]: Yes.

File: src/rpc/mining.cpp (hunk 2 above)
Comment (jamescowens) [reply src/rpc/mining.cpp]: Ok.

**Comment by jamescowens** [comment]
After @barton2526's comments, I decided to take a more aggressive approach to this. It is always a question once you start down this road how far do you go. I have removed (I think) pretty much all externally visible references to "mining" in the GUI and RPC and replaced with staking or crunching as appropriate. Note that I have not made an attempt to rename GRC::Mining... as that is much more invasive.

**Comment by RoboticMind** [comment]
We should probably notify others if we're going to change some of the keys the command uses in its response since that might break some scripts

**Comment by div72** [comment]
> We should probably notify others if we're going to change some of the keys the command uses in its response since that might break some scripts

Even better, let's not touch the keys right now. We can do it in Kermit's Mom. (Sorry Kermit.)

**Comment by jamescowens** [comment]
So the consensus is that I revert the key name changes in the RPC returns (only)?

**Comment by RoboticMind** [comment]
Yes, I think that would be wise. All the other changes aren't likely to break scripts since `getmininginfo` is still an alias. Maybe can make a mention in the Janice release notes about how it will be changed in the release after it, so at least some people are aware.

**Comment by jamescowens** [comment]
Ok!

**Comment by jamescowens** [comment]
Done.

**Comment by jamescowens** [comment]
@div72 This one is ready for review.

**Review by div72 — APPROVED** [review]
ACK 712edb3239fac018e475420dd9424ad6ab1e2450.

**Pull request merged**